nixos/modules/services/web-apps/rutorrent.nix

```diff
@@ -0,0 +1,281 @@
+
+                $XMLRPCMountPoint = "/RPC2";		// DO NOT DELETE THIS LINE!!! DO NOT COMMENT THIS LINE!!!
+
+                $pathToExternals = array(
+                  "php" 	=> ''',			// Something like /usr/bin/php. If empty, will be found in PATH.
+                  "curl"	=> ''',			// Something like /usr/bin/curl. If empty, will be found in PATH.
+                  "gzip"	=> ''',			// Something like /usr/bin/gzip. If empty, will be found in PATH.
+                  "id"	=> ''',			// Something like /usr/bin/id. If empty, will be found in PATH.
+                  "stat"	=> ''',			// Something like /usr/bin/stat. If empty, will be found in PATH.
+                );
+
+                $localhosts = array( 			// list of local interfaces
+                  "127.0.0.1",
+                  "localhost",
+                );
+
+                $profilePath = '../share';		// Path to user profiles
+                $profileMask = 0777;			// Mask for files and directory creation in user profiles.
+                // Both Webserver and rtorrent users must have read-write access to it.
+                // For example, if Webserver and rtorrent users are in the same group then the value may be 0770.
+
+                $tempDirectory = null;			// Temp directory. Absolute path with trail slash. If null, then autodetect will be used.
+
+                $canUseXSendFile = false;		// If true then use X-Sendfile feature if it exist
+
+                $locale = "UTF8";
+            '';
+          in {
+            wantedBy = [ "multi-user.target" ];
+            before = [ "phpfpm-rutorrent.service" ];
+            script = ''
+              ln -sf ${pkgs.rutorrent}/{css,images,js,lang,index.html} ${cfg.home}/
+              mkdir -p ${cfg.home}/{conf,logs,plugins} ${cfg.home}/share/{settings,torrents,users}
+              ln -sf ${pkgs.rutorrent}/conf/{access.ini,plugins.ini} ${cfg.home}/conf/
+              ln -sf ${rutorrentConfig} ${cfg.home}/conf/config.php
+
+              cp -r ${pkgs.rutorrent}/php ${cfg.home}/
+
+              ${optionalString (cfg.plugins != [])
+                ''cp -r ${concatMapStringsSep " " (p: "${pkgs.rutorrent}/plugins/${p}") cfg.plugins} ${cfg.home}/plugins/''}
+
+              chown -R rutorrent:rutorrent ${cfg.home}/{conf,share,logs,plugins}
+              chmod -R 755 ${cfg.home}/{conf,share,logs,plugins}
+            '';
+            serviceConfig.Type = "oneshot";
+          };
+        };
+
+        tmpfiles.rules = [ "d '${cfg.home}' 0775 rutorrent rutorrent -" ];
+      };
+
+      users.groups.rutorrent = {};
+
+      users.users = {
+        rutorrent = {
+          home = cfg.home;
+          group = "rutorrent";
+          extraGroups = [ config.services.rtorrent.group ];
+          description = "rutorrent Daemon user";
+          isSystemUser = true;
+        };
+
+        "${config.services.rtorrent.user}" = {
+          extraGroups = [ "rutorrent" ];
+        };
```

Comment:
I guess this is needed if you manage the filesystem with ruTorrent? So rtorrent still has access to it?

Hm, personally I would run ruTorrent and rtorrent as the same user because otherwise it's gonna create an owner mess on the filesystem. Though I wonder if that would be a good default for NixOS too.

Reply:
Yes, that would be better.

Reply:
Actually I was testing this and I didn't notice anything weird going on with file ownership. It seems rutorrent tries to do as much as possible through rtorrent with XMLRPC. So I think it's actually fine to have a separate user as long as they're in each other's groups. Though in that case it would be nice to have the user and group configurable.